crates/matrix-sdk-ui/src/timeline/controller/mod.rs

```diff
@@ -145,8 +216,11 @@ impl<P: RoomDataProvider> TimelineFocusKind<P> {
     /// Whether to hide in-thread events from the timeline.
     fn hide_threaded_events(&self) -> bool {
         match self {
-            TimelineFocusKind::Live { hide_threaded_events }
-            | TimelineFocusKind::Event { hide_threaded_events, .. } => *hide_threaded_events,
+            TimelineFocusKind::Live { hide_threaded_events } => *hide_threaded_events,
+            TimelineFocusKind::Event { paginator } => paginator
+                .get()
+                .map(|paginator| paginator.hide_threaded_events())
+                .unwrap_or_default(),
```

Comment:
I'm not a fan to use "default" for booleans, let's explicitly say "false":

```suggestion
                .map_or(false, |paginator| paginator.hide_threaded_events())
```